doc/mgmt/Management Framework.md

```diff
@@ -1922,7 +1925,87 @@ The table-transformer is inherited to all descendant nodes unless another one is
 Typically used to check the “when” condition to validate YANG node among multiple nodes to choose only valid nodes from sibling nodes.
+12. `sonic-ext:get-namespace [function]`:
+Multiple DB extension, populates associated namespaces with the YANG path
+Used by Translib service to get all DB's associated with the YANG path. In case of single DB, this extension shall not be used.
+Namespace yang extension shall allow override capability to allow product specific customizations 
+```Sample annotation
+get-namespace annotation sample for openconfig-optical-amplifier yang support: 
+
+Annotation YANG File:
+sonic-mgmt-common/models/yang/annotations/openconfig-optical-amplifier-annot.yang
+module openconfig-optical-amplifier-annot {
+  .....
+    deviation /oc-opt-amp:optical-amplifier/oc-opt-amp:amplifiers/oc-opt-amp:amplifier {
+      deviate add {
+        sonic-ext:key-transformer "oc_name_key_xfmr";
+        sonic-ext:get-namespace "oc_name_get_namespace_xfmr";
+      }
+    }
+  .....
+    deviation /oc-opt-amp:optical-amplifier/oc-opt-amp:amplifiers/oc-opt-amp:amplifier/oc-opt-amp:state {
+      deviate add {
+        sonic-ext:db-name "STATE_DB";
+        sonic-ext:table-name "AMPLIFIER";
+        sonic-ext:key-transformer "oc_name_key_xfmr";
+        sonic-ext:get-namespace "oc_name_get_namespace_xfmr";
+      }
+    }
+  .....
+    deviation /oc-opt-amp:optical-amplifier/oc-opt-amp:supervisory-channels/oc-opt-amp:supervisory-channel {
+      deviate add {
+        sonic-ext:key-transformer "osc_key_xfmr";
+        sonic-ext:get-namespace "oc_name_get_namespace_xfmr";
+      }
+    }
+  .....
+    deviation /oc-opt-amp:optical-amplifier/oc-opt-amp:supervisory-channels/oc-opt-amp:supervisory-channel/oc-opt-amp:state {
+      deviate add {
+        sonic-ext:db-name "STATE_DB";
+        sonic-ext:table-name "OSC";
+        sonic-ext:key-transformer "osc_key_xfmr";
+        sonic-ext:get-namespace "oc_name_get_namespace_xfmr";
+      }
+    }
+   .....
+}
+
+Annotation implementation function
+sonic-mgmt-common/translib/transformer/xfmr_oa.go
+
+func init () {
+ .....
+    XlateFuncBind("oc_name_get_namespace_xfmr", oc_name_get_namespace_xfmr)
+}
+
+var oc_name_get_namespace_xfmr GetNamespaceFunc = func(inParams XfmrParams) ([]string, error) {
+        var response []string
+        var err error
+        var ockey string
+
+        pathInfo := NewPathInfo(inParams.uri)
+
+        if strings.Contains(inParams.uri, "/amplifier") {
+                ockey = pathInfo.Var("name")
 
+        } else if strings.Contains(inParams.uri, "/supervisory-channel") {
+                ockey = pathInfo.Var("interface")
+        }
+
+        // If Key is present in the xpath add the associated namespace and return
+        if ockey != "" {
+		        //db.GetMDBNameFromEntity converts the slot number in the key to namespace
+                dbName := db.GetMDBNameFromEntity(ockey) 
```

Comment:
How is the context of the NB/request URI and the Redis DB table this list maps to being identified in the generic function, GetMDBNameFromEntity? There can be multiple oc lists with same key name.

Reply:
> How is the context of the NB/request URI and the Redis DB table this list maps to being identified in the generic function, GetMDBNameFromEntity? There can be multiple oc lists with same key name.

GetMDBNameFromEntity is a generic function operating on the string passed as input, which will determine the Namespace. Yang specific namespace annotation will have the logic to get the slot specific string from table key and pass it to GetMDBNameFromEntity.